crates/bevy_time/src/time.rs

```diff
@@ -122,11 +148,29 @@ impl Time {
     }
 
     /// The time from startup to the last update in seconds
+    ///
+    /// If you intend to cast this to an `f32` value, note that this value is monotonically increasing,
+    /// and that its precision as an `f32` will noticeably degrade over time (in a matter of hours).
+    /// If that precision loss is unacceptable, you should use [`Time::seconds_since_startup_f32_wrapped`],
+    /// which will return the time from startup modulo a wrapping period.
```

Comment:
I would avoid writing a long paragraph stating that converting an `f64` to an `f32` introduces precision loss (which is obvious). I would simply notice the reader about the existence of a wrapped version, and very briefly highlight the use case. Something like:

> Consider using `seconds_since_startup_f32_wrapped` if you want to reduce precision loss and you are not interested in the absolute time value.

Reply:
The lack of precision is obvious, the fact that it starts being an issue in a matter of a few hours is not. Which is why it's important to make it clear.

Reply:
I agree anyone can understand `f32` has less precision than `f64`, but the part where that precision loss gets worse over time isn't a well-known fact and not something you'd really guess without a understanding of how floating-point numbers work.

Reply:
Fair point. I agree that discerning when wrapping should be done is not trivial. However, the whole type conversion thing seems very forced and use-case specific. What I would do instead is putting some information on the struct level. First I would introduce `seconds_since_startup`. Then specify that in some cases a more floating point precise value is preferred over the absolute time since startup, and then present the wrapped time version.